src/vivarium_nih_us_cvd/components/treatment.py

```diff
@@ -720,69 +734,57 @@ def calculate_initial_medication_coverage_probabilities(
-        ] = sim_start - self.randomness.get_draw(
-            index=sbp_medicated_idx,
-            additional_key="initialized_sbp_medication_start_date",
-        ) * pd.Timedelta(
-            days=365.25 * data_values.INITIALIZATION_MEDICATON_START_DATE_YEARS_IN_PAST
-        )
-
-        ## ldlc medication
-        ldlc_medicated_idx = pop[
-            pop[data_values.COLUMNS.LDLC_MEDICATION]
-            != data_values.LDLC_MEDICATION_LEVEL.NO_TREATMENT.DESCRIPTION
-        ].index
+        medicated_idx = pop[pop[medication_col] != no_treatment_description].index
         pop.loc[
-            ldlc_medicated_idx,
-            data_values.COLUMNS.LDLC_MEDICATION_START_DATE,
+            medicated_idx,
+            start_date_col,
         ] = sim_start - self.randomness.get_draw(
-            index=ldlc_medicated_idx,
-            additional_key="initialized_ldlc_medication_start_date",
+            index=medicated_idx,
+            additional_key=f"initialize_{start_date_col}",
         ) * pd.Timedelta(
             days=365.25 * data_values.INITIALIZATION_MEDICATON_START_DATE_YEARS_IN_PAST
         )
 
         # Initialize medication discontinuation
-        ## sbp medication
-        not_sbp_medicated_idx = pop.index.difference(sbp_medicated_idx)
-        discontinued_sbp_idx = self.randomness.filter_for_probability(
-            population=not_sbp_medicated_idx,
-            probability=data_values.MEDICATION_DISCONTINUATION_PROBABILITY,
-            additional_key="initialize_discontinued_sbp_medication",
-        )
-        pop.loc[discontinued_sbp_idx, data_values.COLUMNS.DISCONTINUED_SBP_MEDICATION] = True
-
-        ## ldlc medication
-        not_ldlc_medicated_idx = pop.index.difference(ldlc_medicated_idx)
-        discontinued_ldlc_idx = self.randomness.filter_for_probability(
-            population=not_ldlc_medicated_idx,
-            probability=data_values.MEDICATION_DISCONTINUATION_PROBABILITY,
-            additional_key="initialize_discontinued_ldlc_medication",
+        not_medicated_idx = pop.index.difference(medicated_idx)
+        # NOTE: The raw discontinuation "relative risk" values were scaled by RT such 
+        # that the scaled probabilities that get calculated in the Treatment component
+        # are <= ~0.95. Ensure future data updates guarantee this as well or
+        # the issue is otherwise handled.
+        scaling_factor = self.medication_coverage_scaling_factors(not_medicated_idx)[
+            {
+                data_values.COLUMNS.SBP_MEDICATION: "sbp_rr",
+                data_values.COLUMNS.LDLC_MEDICATION: "ldl_rr",
+            }[medication_col]
+        ]
+        # NOTE: Unlike medication coverage, we divide the probability by the scaling
+        # factors for discontinuation. This can results in a probability > 1.0
+        # which we clip at 1.0.
+        probs = (data_values.MEDICATION_DISCONTINUATION_PROBABILITY / scaling_factor).clip(
+            upper=1.0
```

Comment:
Isn't this supposed to be clipped at 0.95?

Reply:
Not quite. Syl adjusted the raw data scaling factors such that the calculated probabilities max out at ~0.95. So technically I guess I could remove this clip.